### PR TITLE
Game loop v0

### PR DESCRIPTION
**Pull request by nashirj**
Created basic structure for game loop. Based code off [raspberry turk](http://www.raspberryturk.com and my [brainstorming doc](https://user-images.githubusercontent.com/20624069/141179538-3b42459c-5d8f-4a1a-8eba-c16e9539b653.png)

Once Arduino communication is implemented, this should be a good baseline for the game loop on the raspberry pi.

I included everyone I thought might want to take a look as reviewers. I was mostly hoping for @chasesza to give some input as to whether this seems like it will be compatible with the Arduino code they are writing. If anyone has comments/concerns, please lmk, otherwise I will go ahead and merge this in.

### CODE REVIEW

File: firmware/raspi/game.py [hunk 1]
```diff
@@ -1,101 +1,112 @@
...
+                    board.make_move(move)
+                    # TODO: This is just so we have game loop working, remove once we read from arduino
+                    board.set_status_from_arduino(ArduinoStatus.ExecutingMove)
+                else:
+                    # TODO: do error handling
+                    raise NotImplementedError("Need to handle case of invalid move input. Should we loop until "
+                                              "move is valid? What if the board is messed up? Need to revisit")
+            else:
+                raise ValueError("Other modes of interaction are unimplemented")
+        elif board.get_status_from_arduino() == ArduinoStatus.WaitingForAIMove:
```
[the remark this answers is not in the document]
Comment (ryPattillo) [reply firmware/raspi/game.py:93]: I agree, I was thinking that the Arduino would just be responsible for moving the pieces and will just need to be able to let the PI know if when it is done making a move and if there are any issues. The game loop of the pi should deal with keep track of AI move vs person move , etc..

File: firmware/raspi/game.py [hunk 2]
```diff
@@ -1,101 +1,112 @@
...
+    ai_player = StockfishPlayer()
+
+    # main game loop
+    while True:
+        board_status = board.get_status_from_arduino()
+        print(board_status)
+        # At program start, Arduino is waiting first instruction;
+        # need to have some information passed about who moves first
+        # TODO: this section of code should only 
+        if board_status == ArduinoStatus.WaitingInitialization:
```
[the remark this answers is not in the document]
Comment (LSaber36) [reply firmware/raspi/game.py:58]: I agree with u for sure. The microcontroller isn't doing a whole lot of work really, just needs simple commands and such.

File: firmware/raspi/game.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (LSaber36) [reply firmware/raspi/game.py:93]: For sure, that's probably the better way of doing it. Keeping it simple, which works in this case, will reduce places for errors too.

**Comment by LSaber36** [comment]
I'd suggest looking at the changes, then we're good.

**Pull request merged**